lib/Bitcode/Reader/BitstreamReader.cpp

```diff
@@ -210,9 +266,60 @@ unsigned BitstreamCursor::readRecord(unsigned AbbrevID,
           EltEnc.getEncoding() == BitCodeAbbrevOp::Blob)
         report_fatal_error("Array element type can't be an Array or a Blob");
 
+#if 1 // HLSL Change
+      // Read all the elements a little faster.
+      {
+        BitCodeAbbrevOp::Encoding enc = EltEnc.getEncoding();
+        uint64_t encData = 0;
+        if (EltEnc.hasEncodingData())
+          encData = EltEnc.getEncodingData();
+        unsigned size = (unsigned)encData;
+
+        if (enc == BitCodeAbbrevOp::Fixed) {
+          assert((unsigned)encData <= MaxChunkSize);
+          if (encData == 8 && Uint8Vals) {
+            // Special optimization for fixed elements that are 8 bits
+            Uint8Vals->resize(NumElts);
+            uint8_t *ptr = Uint8Vals->data();
+            unsigned i = 0;
+            for (; NumElts >= 8; NumElts -= 8) {
+              const size_t e = Read(64);
```

Comment:
Does this rely on `sizeof(size_t) == 8`?  This isn't true for 32-bit targets (which we still build).  Should probably use `uint64_t` instead.

Reply:
I removed all explicit magic numbers on the size of the word here. It's now going to read 4 bytes at a time on 32-bit and 8 bytes at a time on 64-bit.